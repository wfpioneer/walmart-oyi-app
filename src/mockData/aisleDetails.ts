import { AisleItem } from '../models/LocationItem';

export const mockAisles: AisleItem[] = [
  {
    aisleId: 1,
    aisleName: 'Aisle 1',
    sectionCount: 30
  },
  {
    aisleId: 2,
    aisleName: 'Aisle 2',
    sectionCount: 0
  },
  {
    aisleId: 3,
    aisleName: 'Aisle 3',
    sectionCount: 2
  },
  {
    aisleId: 4,
    aisleName: 'Aisle 4',
    sectionCount: 2
  },
  {
    aisleId: 5,
    aisleName: 'Aisle 5',
    sectionCount: 30
  },
  {
    aisleId: 6,
    aisleName: 'Aisle 6',
    sectionCount: 0
  }
];
