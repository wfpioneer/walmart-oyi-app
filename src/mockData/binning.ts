export const mockPallets = [{
  id: 123456,
  expirationDate: '10/3/2022',
  firstItem: {
    itemDesc: 'itemDesc',
    price: 123,
    upcNbr: '12343534',
    itemNbr: 351231,
    quantity: 2
  }
}, {
  id: 345345,
  expirationDate: '10/3/2022',
  firstItem: {
    itemDesc: 'itemDesc',
    price: 123,
    upcNbr: '345345',
    itemNbr: 351231,
    quantity: 2
  }
}];
