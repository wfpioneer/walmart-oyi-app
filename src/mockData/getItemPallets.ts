import { AsyncState } from '../models/AsyncState';

export const itemPallets = {
  itemNbr: 987654321,
  upcNbr: '777555333',
  category: 99,
  pallets: [{
    palletId: '4598',
    quantity: 22,
    sectionId: 5578,
    locationName: 'D1-4',
    mixedPallet: false
  }]
};

export const mockGetItemPalletsAsyncState : AsyncState = {
  isWaiting: false,
  value: null,
  result: {
    status: 200,
    data: itemPallets
  },
  error: null
};
