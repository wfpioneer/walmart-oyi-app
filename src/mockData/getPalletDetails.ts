import { ReserveDetailsPallet } from '../models/LocationItems';
import { mockLocationDetails } from './locationDetails';

export interface ReservePallet extends ReserveDetailsPallet {
  palletId: number;
}
const CREATION_DATE = '2022-01-05T12:04:50.000Z';
const EXPIRATION_DATE = '12/31/2022';
export const mockPalletDetails: ReserveDetailsPallet[] = [
  {
    id: 1,
    createDate: CREATION_DATE,
    expirationDate: EXPIRATION_DATE,
    items: [
      {
        itemNbr: 500252,
        itemDesc: 'Nature Valley Honey Nut Granola Bars',
        price: 0,
        upcNbr: '750500252570',
        locationType: 8
      },
      {
        itemNbr: 123456,
        itemDesc: 'Cabbage',
        price: 0,
        upcNbr: '123456000789',
        locationType: 8
      }
    ],
    statusCode: 200
  },
  {
    id: 123,
    createDate: CREATION_DATE,
    expirationDate: EXPIRATION_DATE,
    items: [
      {
        itemNbr: 456789,
        itemDesc: 'Nature Valley Granola Bars',
        price: 1298.0,
        upcNbr: '456789000123',
        locationType: 8
      }
    ],
    statusCode: 200
  },
  {
    id: 456,
    createDate: CREATION_DATE,
    expirationDate: EXPIRATION_DATE,
    items: [
      {
        itemNbr: 123789,
        itemDesc: 'MM电池5号24粒.',
        price: 29.9,
        upcNbr: '123789000456',
        locationType: 8
      }
    ],
    statusCode: 200
  }
];

export const mockCombinedReserveData: ReservePallet[] = [
  { ...mockPalletDetails[0], ...mockLocationDetails.pallets.palletData[0] },
  { ...mockPalletDetails[1], ...mockLocationDetails.pallets.palletData[1] },
  { ...mockPalletDetails[2], ...mockLocationDetails.pallets.palletData[2] }
];
