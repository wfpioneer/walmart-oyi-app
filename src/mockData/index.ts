
import itemDetails from './getItemDetails';


export const getMockItemDetails = (upc: string) => {
  // @ts-ignore
  return itemDetails[upc] || {};
};
