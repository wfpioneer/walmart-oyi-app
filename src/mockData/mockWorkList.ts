import { MissingPalletWorklistItemI, WorklistItemI } from '../models/WorklistItem';
import { FilteredCategory } from '../models/FilterListItem';

// eslint-disable-next-line no-shadow
enum ITEM_NAME {
  ELECTRONIC_ITEM = 'ELECTRONIC ITEM',
  BAKERY_ITEM = 'BAKERY ITEM',
  WINE_ITEM = 'WINE ITEM',
  PHARMACY_ITEM = 'PHARMACY ITEM',
  PHARMACY_ITEM_1 = 'PHARMACY ITEM 1',
  TEST_ITEM = 'TEST ITEM'
}

// eslint-disable-next-line no-shadow
enum CATEGORY_NAME {
  FOODSERVICE = 'FOODSERVICE',
  ELECTRONICS = 'ELECTRONICS',
  FRESH_BAKERY = 'FRESH BAKERY',
  WINE = 'WINE',
  PHARMACY_RX = 'PHARMACY RX'
}

// eslint-disable-next-line no-shadow
enum WORKLISTTYPE {
  NSFL = 'NSFL',
  C = 'C',
  NO = 'NO',
  NS = 'NS',
  CATEGORY = 'CATEGORY',
  NON_EXCEPTION = 'Non-Exception',
  MP = 'MP'
}

const COMPLETED_TS = '2020-07-15T08:02:17-05:00';

export const mockWorkListToDo: WorklistItemI[] = [
  {
    worklistType: WORKLISTTYPE.NSFL,
    itemName: ITEM_NAME.TEST_ITEM,
    itemNbr: 1234567890,
    upcNbr: '000055559999',
    catgNbr: 93,
    catgName: CATEGORY_NAME.FOODSERVICE,
    subCatgNbr: 0,
    subCatgName: undefined,
    completedTs: undefined,
    completedUserId: undefined,
    completed: false
  },
  {
    worklistType: WORKLISTTYPE.NO,
    itemName: ITEM_NAME.ELECTRONIC_ITEM,
    itemNbr: 987654321,
    upcNbr: '777555333',
    catgNbr: 99,
    catgName: CATEGORY_NAME.ELECTRONICS,
    subCatgNbr: 0,
    subCatgName: undefined,
    completedTs: undefined,
    completedUserId: undefined,
    completed: false
  },
  {
    worklistType: WORKLISTTYPE.NO,
    itemName: ITEM_NAME.BAKERY_ITEM,
    itemNbr: 123789456,
    upcNbr: '111122223333',
    catgNbr: 88,
    catgName: CATEGORY_NAME.FRESH_BAKERY,
    subCatgNbr: 0,
    subCatgName: undefined,
    completedTs: undefined,
    completedUserId: undefined,
    completed: false
  },
  {
    worklistType: WORKLISTTYPE.C,
    itemName: ITEM_NAME.WINE_ITEM,
    itemNbr: 456789123,
    upcNbr: '444455556666',
    catgNbr: 19,
    catgName: CATEGORY_NAME.WINE,
    subCatgNbr: 0,
    subCatgName: undefined,
    completedTs: undefined,
    completedUserId: undefined,
    completed: false
  },
  {
    worklistType: WORKLISTTYPE.C,
    itemName: ITEM_NAME.PHARMACY_ITEM,
    itemNbr: 789123456,
    upcNbr: '777788889999',
    catgNbr: 87,
    catgName: CATEGORY_NAME.PHARMACY_RX,
    subCatgNbr: 0,
    subCatgName: undefined,
    completedTs: undefined,
    completedUserId: undefined,
    completed: false
  },
  {
    worklistType: WORKLISTTYPE.NS,
    itemName: ITEM_NAME.PHARMACY_ITEM_1,
    itemNbr: 789123457,
    upcNbr: '667788889999',
    catgNbr: 199,
    catgName: CATEGORY_NAME.ELECTRONICS,
    subCatgNbr: 0,
    subCatgName: undefined,
    completedTs: undefined,
    completedUserId: undefined,
    completed: false
  }
];
export const mockWorkListComplete: WorklistItemI[] = [
  {
    worklistType: WORKLISTTYPE.NSFL,
    itemName: ITEM_NAME.TEST_ITEM,
    itemNbr: 1234567890,
    upcNbr: '000055559999',
    catgNbr: 93,
    catgName: CATEGORY_NAME.FOODSERVICE,
    subCatgNbr: 0,
    subCatgName: undefined,
    completedTs: COMPLETED_TS,
    completedUserId: 'SVCintlsamsops',
    completed: true
  },
  {
    worklistType: WORKLISTTYPE.NO,
    itemName: ITEM_NAME.ELECTRONIC_ITEM,
    itemNbr: 987654321,
    upcNbr: '777555333',
    catgNbr: 99,
    catgName: CATEGORY_NAME.ELECTRONICS,
    subCatgNbr: 0,
    subCatgName: undefined,
    completedTs: COMPLETED_TS,
    completedUserId: 'SVCintlsamsops',
    completed: true
  },
  {
    worklistType: WORKLISTTYPE.NO,
    itemName: ITEM_NAME.BAKERY_ITEM,
    itemNbr: 123789456,
    upcNbr: '111122223333',
    catgNbr: 88,
    catgName: CATEGORY_NAME.FRESH_BAKERY,
    subCatgNbr: 0,
    subCatgName: undefined,
    completedTs: COMPLETED_TS,
    completedUserId: 'SVCintlsamsops',
    completed: true
  },
  {
    worklistType: WORKLISTTYPE.C,
    itemName: ITEM_NAME.WINE_ITEM,
    itemNbr: 456789123,
    upcNbr: '444455556666',
    catgNbr: 19,
    catgName: CATEGORY_NAME.WINE,
    subCatgNbr: 0,
    subCatgName: undefined,
    completedTs: COMPLETED_TS,
    completedUserId: 'SVCintlsamsops',
    completed: true
  },
  {
    worklistType: WORKLISTTYPE.C,
    itemName: ITEM_NAME.PHARMACY_ITEM,
    itemNbr: 789123456,
    upcNbr: '777788889999',
    catgNbr: 87,
    catgName: CATEGORY_NAME.PHARMACY_RX,
    subCatgNbr: 0,
    subCatgName: undefined,
    completedTs: COMPLETED_TS,
    completedUserId: 'SVCintlsamsops',
    completed: true
  },
  {
    worklistType: WORKLISTTYPE.NS,
    itemName: ITEM_NAME.PHARMACY_ITEM_1,
    itemNbr: 789123457,
    upcNbr: '667788889999',
    catgNbr: 199,
    catgName: CATEGORY_NAME.ELECTRONICS,
    subCatgNbr: 0,
    subCatgName: undefined,
    completedTs: undefined,
    completedUserId: undefined,
    completed: true
  }
];

export const mockCategoryList: WorklistItemI[] = [
  {
    worklistType: WORKLISTTYPE.CATEGORY,
    catgName: CATEGORY_NAME.WINE,
    catgNbr: 19,
    itemCount: 1
  },
  {
    worklistType: WORKLISTTYPE.C,
    itemName: ITEM_NAME.WINE_ITEM,
    itemNbr: 456789123,
    upcNbr: '444455556666',
    catgNbr: 19,
    catgName: CATEGORY_NAME.WINE,
    subCatgNbr: 0,
    subCatgName: undefined,
    completedTs: undefined,
    completedUserId: undefined,
    completed: false
  },
  {
    worklistType: WORKLISTTYPE.CATEGORY,
    catgName: CATEGORY_NAME.PHARMACY_RX,
    catgNbr: 87,
    itemCount: 1
  },
  {
    worklistType: WORKLISTTYPE.C,
    itemName: ITEM_NAME.PHARMACY_ITEM,
    itemNbr: 789123456,
    upcNbr: '777788889999',
    catgNbr: 87,
    catgName: CATEGORY_NAME.PHARMACY_RX,
    subCatgNbr: 0,
    subCatgName: undefined,
    completedTs: undefined,
    completedUserId: undefined,
    completed: false
  },
  {
    worklistType: WORKLISTTYPE.CATEGORY,
    catgName: CATEGORY_NAME.FRESH_BAKERY,
    catgNbr: 88,
    itemCount: 1
  },
  {
    worklistType: WORKLISTTYPE.NO,
    itemName: ITEM_NAME.BAKERY_ITEM,
    itemNbr: 123789456,
    upcNbr: '111122223333',
    catgNbr: 88,
    catgName: CATEGORY_NAME.FRESH_BAKERY,
    subCatgNbr: 0,
    subCatgName: undefined,
    completedTs: undefined,
    completedUserId: undefined,
    completed: false
  },
  {
    worklistType: WORKLISTTYPE.CATEGORY,
    catgName: CATEGORY_NAME.FOODSERVICE,
    catgNbr: 93,
    itemCount: 1
  },
  {
    worklistType: WORKLISTTYPE.NSFL,
    itemName: ITEM_NAME.TEST_ITEM,
    itemNbr: 1234567890,
    upcNbr: '000055559999',
    catgNbr: 93,
    catgName: CATEGORY_NAME.FOODSERVICE,
    subCatgNbr: 0,
    subCatgName: undefined,
    completedTs: undefined,
    completedUserId: undefined,
    completed: false
  },
  {
    worklistType: WORKLISTTYPE.CATEGORY,
    catgName: CATEGORY_NAME.ELECTRONICS,
    catgNbr: 99,
    itemCount: 1
  },
  {
    worklistType: WORKLISTTYPE.NO,
    itemName: ITEM_NAME.ELECTRONIC_ITEM,
    itemNbr: 987654321,
    upcNbr: '777555333',
    catgNbr: 99,
    catgName: CATEGORY_NAME.ELECTRONICS,
    subCatgNbr: 0,
    subCatgName: undefined,
    completedTs: undefined,
    completedUserId: undefined,
    completed: false
  },
  {
    worklistType: WORKLISTTYPE.CATEGORY,
    catgName: CATEGORY_NAME.ELECTRONICS,
    catgNbr: 199,
    itemCount: 1
  },
  {
    worklistType: WORKLISTTYPE.NS,
    itemName: ITEM_NAME.PHARMACY_ITEM_1,
    itemNbr: 789123457,
    upcNbr: '667788889999',
    catgNbr: 199,
    catgName: CATEGORY_NAME.ELECTRONICS,
    subCatgNbr: 0,
    subCatgName: undefined,
    completedTs: undefined,
    completedUserId: undefined,
    completed: false
  }
];

export const missingCategoryNbrList = [{
  worklistType: WORKLISTTYPE.NSFL,
  itemName: ITEM_NAME.TEST_ITEM,
  itemNbr: 1234567890,
  upcNbr: '000055559999',
  catgNbr: undefined,
  catgName: CATEGORY_NAME.FOODSERVICE,
  subCatgNbr: 0,
  subCatgName: undefined,
  completedTs: undefined,
  completedUserId: undefined,
  completed: false
},
{
  worklistType: WORKLISTTYPE.NO,
  itemName: ITEM_NAME.ELECTRONIC_ITEM,
  itemNbr: 987654321,
  upcNbr: '777555333',
  catgNbr: undefined,
  catgName: CATEGORY_NAME.ELECTRONICS,
  subCatgNbr: 0,
  subCatgName: undefined,
  completedTs: undefined,
  completedUserId: undefined,
  completed: false
},
{
  worklistType: WORKLISTTYPE.NO,
  itemName: ITEM_NAME.BAKERY_ITEM,
  itemNbr: 123789456,
  upcNbr: '111122223333',
  catgNbr: undefined,
  catgName: CATEGORY_NAME.FRESH_BAKERY,
  subCatgNbr: 0,
  subCatgName: undefined,
  completedTs: undefined,
  completedUserId: undefined,
  completed: false
},
{
  worklistType: WORKLISTTYPE.C,
  itemName: ITEM_NAME.WINE_ITEM,
  itemNbr: 456789123,
  upcNbr: '444455556666',
  catgNbr: undefined,
  catgName: CATEGORY_NAME.WINE,
  subCatgNbr: 0,
  subCatgName: undefined,
  completedTs: undefined,
  completedUserId: undefined,
  completed: false
},
{
  worklistType: WORKLISTTYPE.C,
  itemName: ITEM_NAME.PHARMACY_ITEM,
  itemNbr: 789123456,
  upcNbr: '777788889999',
  catgNbr: undefined,
  catgName: CATEGORY_NAME.PHARMACY_RX,
  subCatgNbr: 0,
  subCatgName: undefined,
  completedTs: undefined,
  completedUserId: undefined,
  completed: false
}];

export const missingExceptionsWorklist: WorklistItemI[] = [
  {
    worklistType: WORKLISTTYPE.NON_EXCEPTION,
    itemName: ITEM_NAME.TEST_ITEM,
    itemNbr: 1234567890,
    upcNbr: '000055559999',
    catgNbr: 93,
    catgName: CATEGORY_NAME.FOODSERVICE,
    subCatgNbr: 0,
    subCatgName: undefined,
    completedTs: COMPLETED_TS,
    completedUserId: 'SVCintlsamsops',
    completed: true
  },
  {
    worklistType: WORKLISTTYPE.NON_EXCEPTION,
    itemName: ITEM_NAME.ELECTRONIC_ITEM,
    itemNbr: 987654321,
    upcNbr: '777555333',
    catgNbr: 99,
    catgName: CATEGORY_NAME.ELECTRONICS,
    subCatgNbr: 0,
    subCatgName: undefined,
    completedTs: COMPLETED_TS,
    completedUserId: 'SVCintlsamsops',
    completed: true
  },
  {
    worklistType: WORKLISTTYPE.NON_EXCEPTION,
    itemName: ITEM_NAME.BAKERY_ITEM,
    itemNbr: 123789456,
    upcNbr: '111122223333',
    catgNbr: 88,
    catgName: CATEGORY_NAME.FRESH_BAKERY,
    subCatgNbr: 0,
    subCatgName: undefined,
    completedTs: COMPLETED_TS,
    completedUserId: 'SVCintlsamsops',
    completed: true
  },
  {
    worklistType: WORKLISTTYPE.NON_EXCEPTION,
    itemName: ITEM_NAME.PHARMACY_ITEM_1,
    itemNbr: 789123457,
    upcNbr: '667788889999',
    catgNbr: 199,
    catgName: CATEGORY_NAME.ELECTRONICS,
    subCatgNbr: 0,
    subCatgName: undefined,
    completedTs: undefined,
    completedUserId: undefined,
    completed: true
  }
];

export const mockCategoryMap: FilteredCategory[] = [
  {
    catgName: CATEGORY_NAME.WINE,
    catgNbr: 19,
    selected: false
  },
  {
    catgName: CATEGORY_NAME.PHARMACY_RX,
    catgNbr: 87,
    selected: false
  },
  {
    catgName: CATEGORY_NAME.FRESH_BAKERY,
    catgNbr: 88,
    selected: false
  },
  {
    catgName: CATEGORY_NAME.FOODSERVICE,
    catgNbr: 93,
    selected: false
  },
  {
    catgName: CATEGORY_NAME.ELECTRONICS,
    catgNbr: 99,
    selected: false
  },
  {
    catgName: CATEGORY_NAME.ELECTRONICS,
    catgNbr: 199,
    selected: false
  }
];

export const mockMissingPalletWorklistTodo: MissingPalletWorklistItemI[] = [
  {
    createUserId: '11',
    createTs: '26/06/2022',
    lastKnownPalletLocationId: 1,
    lastKnownPalletLocationName: 'A1-1',
    palletDeleted: false,
    palletId: 7988,
    worklistType: 'MP',
    completed: false,
    completedUserId: undefined,
    completedTs: undefined
  },
  {
    createUserId: '12',
    createTs: '26/06/2022',
    lastKnownPalletLocationId: 2,
    lastKnownPalletLocationName: 'A1-2',
    palletDeleted: false,
    palletId: 7989,
    worklistType: 'MP',
    completed: false,
    completedUserId: undefined,
    completedTs: undefined
  }
];

export const mockMissingPalletWorklistComplete: MissingPalletWorklistItemI[] = [
  {
    createUserId: '13',
    createTs: '26/06/2022',
    lastKnownPalletLocationId: 3,
    lastKnownPalletLocationName: 'B1-1',
    palletDeleted: false,
    palletId: 7990,
    worklistType: 'MP',
    completed: true,
    completedUserId: '1',
    completedTs: '2022-07-07T12:40:00.000Z'
  },
  {
    createUserId: '14',
    createTs: '26/06/2022',
    lastKnownPalletLocationId: 4,
    lastKnownPalletLocationName: 'B1-2',
    palletDeleted: false,
    palletId: 7991,
    worklistType: 'MP',
    completed: true,
    completedUserId: '2',
    completedTs: '2022-07-07T12:40:00.000Z'
  }
];
