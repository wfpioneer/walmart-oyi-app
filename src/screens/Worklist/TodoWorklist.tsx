import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { WorklistItemI } from '../../models/WorklistItem';
import { Worklist } from './Worklist';
import { useTypedSelector } from '../../state/reducers/RootReducer';
import { getWorklist } from '../../state/actions/saga';
import { trackEvent } from '../../utils/AppCenterTool';

export const TodoWorklist = () => {
  const { isWaiting, result, error } = useTypedSelector(state => state.async.getWorklist);
  const dispatch = useDispatch();

  useEffect(() => {
    if (result) {
      trackEvent('worklist_items_api_success');
    }

    if (error) {
      trackEvent('worklist_items_api_error', {errorDetails: error.message || error});
    }
  }, [result, error]);

  let todoData: WorklistItemI[] | undefined;

  if (result && result.data) {
    todoData = result.data.filter((item: WorklistItemI) => item.completed === false);
  }

  return (
    <Worklist
      data={todoData}
      refreshing={isWaiting}
      onRefresh={() => dispatch(getWorklist())}
      error={error}
    />
  );
};
