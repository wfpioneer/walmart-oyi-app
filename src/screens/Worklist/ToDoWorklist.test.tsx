import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { TodoWorklistScreen } from './TodoWorklist';
import { mockWorkListComplete, mockWorkListToDo } from '../../mockData/mockWorkList';

jest.mock('../../utils/AppCenterTool', () => jest.requireActual('../../utils/__mocks__/AppCenterTool'));
jest.mock('../../utils/sessionTimeout.ts', () => jest.requireActual('../../utils/__mocks__/sessTimeout'));
describe('ToDoWorklistScreen', () => {
  describe('Tests rendering Worklist data', () => {
    it('Renders array of uncompleted worklist items', () => {
      const renderer = ShallowRenderer.createRenderer();
      const toDoWorklistResult = {
        data: mockWorkListToDo
      };
      renderer.render(
        <TodoWorklistScreen
          isWaiting={false}
          result={toDoWorklistResult}
          error={null}
          dispatch={jest.fn()}
          useEffectHook={jest.fn()}
          filterCategories={[]}
          filterExceptions={[]}
          groupToggle={false}
          updateGroupToggle={jest.fn()}
        />
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
    it('Renders empty array for completed worklist items ', () => {
      const renderer = ShallowRenderer.createRenderer();
      const completeWorklistResult = {
        data: mockWorkListComplete
      };
      renderer.render(
        <TodoWorklistScreen
          isWaiting={false}
          result={completeWorklistResult}
          error={null}
          dispatch={jest.fn()}
          useEffectHook={jest.fn()}
          filterCategories={[]}
          filterExceptions={[]}
          groupToggle={false}
          updateGroupToggle={jest.fn()}
        />
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
    it('Should only render incomplete items for a WorkList with incomplete & complete items', () => {
      const renderer = ShallowRenderer.createRenderer();
      const mixedWorklistResult = {
        data: [...mockWorkListToDo, ...mockWorkListComplete]
      };
      renderer.render(
        <TodoWorklistScreen
          isWaiting={false}
          result={mixedWorklistResult}
          error={null}
          dispatch={jest.fn()}
          useEffectHook={jest.fn()}
          filterCategories={[]}
          filterExceptions={[]}
          groupToggle={false}
          updateGroupToggle={jest.fn()}
        />
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
    it('Renders empty array for zero worklist items', () => {
      const renderer = ShallowRenderer.createRenderer();
      const emptyWorklistResult = {
        data: []
      };
      renderer.render(
        <TodoWorklistScreen
          isWaiting={false}
          result={emptyWorklistResult}
          error={null}
          dispatch={jest.fn()}
          useEffectHook={jest.fn()}
          filterCategories={[]}
          filterExceptions={[]}
          groupToggle={false}
          updateGroupToggle={jest.fn()}
        />
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
  });
  describe('Tests rendering worklist api responses', () => {
    it('Renders error message for failed worklist request', () => {
      const renderer = ShallowRenderer.createRenderer();

      renderer.render(
        <TodoWorklistScreen
          isWaiting={false}
          result={null}
          error="Network Error"
          dispatch={jest.fn()}
          useEffectHook={jest.fn()}
          filterCategories={[]}
          filterExceptions={[]}
          groupToggle={false}
          updateGroupToggle={jest.fn()}
        />
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
    it('Renders refresh set to true waiting for worklist api response', () => {
      const renderer = ShallowRenderer.createRenderer();

      renderer.render(
        <TodoWorklistScreen
          isWaiting={true}
          result={null}
          error={null}
          dispatch={jest.fn()}
          useEffectHook={jest.fn()}
          filterCategories={[]}
          filterExceptions={[]}
          groupToggle={false}
          updateGroupToggle={jest.fn()}
        />
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
  });
});
