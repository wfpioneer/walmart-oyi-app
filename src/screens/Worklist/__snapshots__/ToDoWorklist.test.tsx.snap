// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoWorklistScreen Tests rendering Worklist data Renders array of uncompleted worklist items 1`] = `
<Worklist
  data={
    Array [
      Object {
        "catgName": "FOODSERVICE",
        "catgNbr": 93,
        "completed": false,
        "completedTs": undefined,
        "completedUserId": undefined,
        "itemName": "TEST ITEM",
        "itemNbr": 1234567890,
        "subCatgName": undefined,
        "subCatgNbr": 0,
        "upcNbr": "000055559999",
        "worklistType": "NSFL",
      },
      Object {
        "catgName": "ELECTRONICS",
        "catgNbr": 99,
        "completed": false,
        "completedTs": undefined,
        "completedUserId": undefined,
        "itemName": "ELECTRONIC ITEM",
        "itemNbr": 987654321,
        "subCatgName": undefined,
        "subCatgNbr": 0,
        "upcNbr": "777555333",
        "worklistType": "NO",
      },
      Object {
        "catgName": "FRESH BAKERY",
        "catgNbr": 87,
        "completed": false,
        "completedTs": undefined,
        "completedUserId": undefined,
        "itemName": "BAKERY ITEM",
        "itemNbr": 123789456,
        "subCatgName": undefined,
        "subCatgNbr": 0,
        "upcNbr": "111122223333",
        "worklistType": "NO",
      },
      Object {
        "catgName": "WINE",
        "catgNbr": 19,
        "completed": false,
        "completedTs": undefined,
        "completedUserId": undefined,
        "itemName": "WINE ITEM",
        "itemNbr": 456789123,
        "subCatgName": undefined,
        "subCatgNbr": 0,
        "upcNbr": "444455556666",
        "worklistType": "C",
      },
      Object {
        "catgName": "PHARMACY RX",
        "catgNbr": 87,
        "completed": false,
        "completedTs": undefined,
        "completedUserId": undefined,
        "itemName": "PHARMACY ITEM",
        "itemNbr": 789123456,
        "subCatgName": undefined,
        "subCatgNbr": 0,
        "upcNbr": "777788889999",
        "worklistType": "C",
      },
    ]
  }
  error={null}
  onRefresh={[Function]}
  refreshing={false}
/>
`;

exports[`ToDoWorklistScreen Tests rendering Worklist data Renders empty array for completed worklist items  1`] = `
<Worklist
  data={Array []}
  error={null}
  onRefresh={[Function]}
  refreshing={false}
/>
`;

exports[`ToDoWorklistScreen Tests rendering Worklist data Renders empty array for zero worklist items 1`] = `
<Worklist
  data={Array []}
  error={null}
  onRefresh={[Function]}
  refreshing={false}
/>
`;

exports[`ToDoWorklistScreen Tests rendering Worklist data Should only render incomplete items for a WorkList with incomplete & complete items 1`] = `
<Worklist
  data={
    Array [
      Object {
        "catgName": "FOODSERVICE",
        "catgNbr": 93,
        "completed": false,
        "completedTs": undefined,
        "completedUserId": undefined,
        "itemName": "TEST ITEM",
        "itemNbr": 1234567890,
        "subCatgName": undefined,
        "subCatgNbr": 0,
        "upcNbr": "000055559999",
        "worklistType": "NSFL",
      },
      Object {
        "catgName": "ELECTRONICS",
        "catgNbr": 99,
        "completed": false,
        "completedTs": undefined,
        "completedUserId": undefined,
        "itemName": "ELECTRONIC ITEM",
        "itemNbr": 987654321,
        "subCatgName": undefined,
        "subCatgNbr": 0,
        "upcNbr": "777555333",
        "worklistType": "NO",
      },
      Object {
        "catgName": "FRESH BAKERY",
        "catgNbr": 87,
        "completed": false,
        "completedTs": undefined,
        "completedUserId": undefined,
        "itemName": "BAKERY ITEM",
        "itemNbr": 123789456,
        "subCatgName": undefined,
        "subCatgNbr": 0,
        "upcNbr": "111122223333",
        "worklistType": "NO",
      },
      Object {
        "catgName": "WINE",
        "catgNbr": 19,
        "completed": false,
        "completedTs": undefined,
        "completedUserId": undefined,
        "itemName": "WINE ITEM",
        "itemNbr": 456789123,
        "subCatgName": undefined,
        "subCatgNbr": 0,
        "upcNbr": "444455556666",
        "worklistType": "C",
      },
      Object {
        "catgName": "PHARMACY RX",
        "catgNbr": 87,
        "completed": false,
        "completedTs": undefined,
        "completedUserId": undefined,
        "itemName": "PHARMACY ITEM",
        "itemNbr": 789123456,
        "subCatgName": undefined,
        "subCatgNbr": 0,
        "upcNbr": "777788889999",
        "worklistType": "C",
      },
    ]
  }
  error={null}
  onRefresh={[Function]}
  refreshing={false}
/>
`;

exports[`ToDoWorklistScreen Tests rendering worklist api responses Renders error message for failed worklist request 1`] = `
<Worklist
  error="Network Error"
  onRefresh={[Function]}
  refreshing={false}
/>
`;

exports[`ToDoWorklistScreen Tests rendering worklist api responses Renders refresh set to true waiting for worklist api response 1`] = `
<Worklist
  error={null}
  onRefresh={[Function]}
  refreshing={true}
/>
`;
